extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/cache/polling/OnHeapPollingCache.java

```diff
@@ -83,6 +84,19 @@ public List<K> getKeys(final V value)
     return listOfKeys;
   }
 
+  @Override
+  @SuppressWarnings("unchecked")
+  public long estimateHeapFootprint()
+  {
+    for (final Map.Entry<K, V> entry : immutableMap.entrySet()) {
+      if (!(entry.getKey() instanceof String) || !(entry.getValue() instanceof String)) {
+        return 0;
+      }
+    }
+
+    return MapLookupExtractor.estimateHeapFootprint((Map<String, String>) immutableMap);
```

Comment:
IMHO, Instead of the monitor which runs every 10 mins, the place where we are updating the cache should do the running size calculation. Then this method just becomes a getter on top of that variable. 

We can do that as part of another PR as well.

Reply:
Makes sense to me. I'd rather that be a different PR, especially since I'm not really changing existing behavior in this PR. (Current code in master does the footprint estimation every time the monitor is run.)